src/main/webapp/resources/js/pages/projects/samples-metadata-import/redux/importReducer.ts

```diff
@@ -93,77 +94,145 @@ export const saveMetadata = createAsyncThunk<
         })),
     });
 
-    const chunkSize = 100;
-    for (let i = 0; i < metadata.length; i = i + chunkSize) {
-      const promises: Promise<void>[] = [];
-      for (let j = i; j < i + chunkSize && j < metadata.length; j++) {
-        const metadataItem: MetadataItem = metadata[j];
-        const index: string = metadataItem.rowKey;
-        if (
-          selectedMetadataKeys.includes(index) &&
-          metadataSaveDetails[index]?.saved !== true
-        ) {
-          const name: string = metadataItem[sampleNameColumn];
-          const metadataFields: FieldUpdate[] = Object.entries(metadataItem)
-            .filter(
-              ([key]) =>
-                headers.map((header) => header.name).includes(key) &&
-                key !== sampleNameColumn
-            )
-            .map(([key, value]) => ({
-              field: key,
-              value,
-              restriction: headers.filter((header) => header.name === key)[0]
-                .targetRestriction,
-            }));
-          const sampleId = metadataValidateDetails[index].foundSampleId;
-          if (sampleId) {
-            promises.push(
-              updateSample({
-                projectId,
-                sampleId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          } else {
-            promises.push(
-              createSample({
-                projectId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          }
-        }
-      }
-      await Promise.all(promises).then(() => {
-        dispatch(
-          setMetadataSaveDetails(Object.assign({}, metadataSaveDetails))
+    const selectedSampleList = metadata.filter((metadataItem) => {
+      const name: string = metadataItem[sampleNameColumn];
+      return (
+        selectedMetadataKeys.includes(metadataItem.rowKey) &&
+        metadataSaveDetails[name]?.saved !== true
+      );
+    });
+
+    const chunkSize = 10;
```

Comment:
Can you update this to be a dynamic chunk size like we had discussed. I think we settled on a chunkSize range of 10-100. See https://github.com/phac-nml/irida/blob/development/src/main/webapp/resources/js/pages/projects/linelist/sagas/entries.js#L13 for an example calculation.

May need to do this for updates and creates separately.